pkg/volume/rbd/rbd_util.go

```diff
@@ -293,9 +404,21 @@ func (util *RBDUtil) DetachDisk(plugin *rbdPlugin, deviceMountPath string, devic
 	if len(device) == 0 {
 		return fmt.Errorf("DetachDisk failed , device is empty")
 	}
-	// rbd unmap
+
 	exec := plugin.host.GetExec(plugin.GetPluginName())
-	output, err := exec.Run("rbd", "unmap", device)
+
+	var rbdCmd string
+
+	// Unlike map, we cannot fallthrough for unmap
+	// the tool to unmap is based on device type
+	if strings.HasPrefix(device, "/dev/nbd") {
+		rbdCmd = "rbd-nbd"
+	} else {
+		rbdCmd = "rbd"
+	}
+
+	// rbd unmap
+	output, err := exec.Run(rbdCmd, "unmap", device)
```

Comment:
what happens if you map using rbd, can you unmap using rbd-nbd ? If that is not a supported scenario we should detect and error out

Reply:
No you cannot unmap rbd devices using rbd-nbd, as per comments on L412 & L413. This code detects how the device was mapped, and uses the appropriate command to unmap.